pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -138,6 +143,13 @@ func (cl *Client) RegisterEventHandler(kind config.GroupVersionKind, handler mod
 
 // Run the queue and all informers. Callers should  wait for HasSynced() before depending on results.
 func (cl *Client) Run(stop <-chan struct{}) {
+	cl.mutex.Lock()
```

Comment:
this will block one of them indefinitely if its stuck on wait for cache sync. Maybe we just wantL151 to be

```
 if cl.Started.Swap(true) {
// was already started by other thread
 return
}
```

?

Reply:
good suggestion, done